### PR TITLE
check for insights group and root user

### PR DESCRIPTION
**Pull request by brantleyr**

### CODE REVIEW

File: insights_client/__init__.py [hunk 1]
```diff
@@ -30,13 +31,16 @@
 try:
     insights_uid = pwd.getpwnam("insights").pw_uid
     insights_gid = pwd.getpwnam("insights").pw_gid
+    insights_grpid = grp.getgrpname("insights").gr_gid
+    insights_grpusers = grp.getgrpname("insights").gr_mem
```
Comment (kylape) [inline insights_client/__init__.py]: Is this used anywhere?

File: insights_client/__init__.py [hunk 2]
```diff
@@ -30,13 +31,16 @@
 try:
     insights_uid = pwd.getpwnam("insights").pw_uid
     insights_gid = pwd.getpwnam("insights").pw_gid
+    insights_grpid = grp.getgrpname("insights").gr_gid
```
Comment (kylape) [inline insights_client/__init__.py]: Is this always the same as `pwd.getpwnam("insights").pw_gid`?

File: insights_client/__init__.py [hunk 3]
```diff
@@ -30,13 +31,16 @@
...
 except:
-    insights_uid, insights_gid = None, None
-    raise
+    sys.exit("User and group 'insights' not found. Exiting.")
 
 
 def demote(uid, gid, phase):
-    if uid and gid and phase != "collect":
+    user_is_root = os.geteuid() is 0
+    if not user_is_root:
```
Comment (kylape) [inline insights_client/__init__.py]: Small nitpick - IMHO `if os.geteuid() == 0` is just as readable.

File: insights_client/__init__.py [hunk 4]
```diff
@@ -97,20 +106,34 @@ def _main():
...
     """
 
+    # check for insights user/group
     if not (insights_uid or insights_gid):
         print("WARNING: 'insights' user not found.  Using root to run all phases")
 
+    # check if the user is in the insights group
+    # make sure they are not root
+    in_insights_group = insights_grpid in curr_user_grps
+    is_root = os.geteuid() is 0
```
Comment (kylape) [inline insights_client/__init__.py]: Same nitpick as above

File: insights_client/__init__.py (hunk 2 above)
Comment (brantleyr) [reply insights_client/__init__.py]: I don't think so? I would feel safer if we explicitly got the group ID. But you might be correct.

File: insights_client/__init__.py (hunk 1 above)
Comment (brantleyr) [reply insights_client/__init__.py]: insights_grpusers is not used anywhere so I can remove that.

File: insights_client/__init__.py (hunk 3 above)
Comment (brantleyr) [reply insights_client/__init__.py]: Do you mean `if os.geteuid() != 0`?

File: insights_client/__init__.py (hunk 2 above)
Comment (kylape) [reply insights_client/__init__.py]: Yeah that's fine.

**Pull request merged**